// @chipsgg/openservice-ws-client/src/utils.js
module.exports.decode = function (data) {
  return JSON.parse(data);
};
module.exports.encode = function (data) {
  return JSON.stringify(data);
};
module.exports.isEvent = function (data) {
  return data[1] == null;
};
module.exports.isRpc = function (data) {
  return data[1] != null;
};
const touch = (module.exports.touch = function (
  state = {},
  path = [],
  cb = (x) => x,
  i = 0
) {
  if (path.length == 0) return data;
  if (i == path.length - 1) {
    cb(state, path[i]);
    return state;
  }
  const head = path[i];
  state[head] = { ...state[head] };
  touch(state[head], path, cb, i + 1);
  return state;
});
module.exports.set = function (state, path, data) {
  return touch(state, path, (obj, key) => (obj[key] = data));
};
const unset = (module.exports.unset = function (
  state = {},
  path = [],
  cb = (obj, key) => delete obj[key],
  i = 0
) {
  if (path.length == 0) return data;
  if (i == path.length - 1) {
    cb(state, path[i]);
    return state;
  }
  const head = path[i];
  if (state[head] == null) {
    return state;
  } else {
    state[head] = { ...state[head] };
  }
  unset(state[head], path, cb, i + 1);
  return state;
});
