// @chipsgg/openservice-ws-client/rollup.config.js
import auto from "@rollup/plugin-auto-install";
import resolve from "@rollup/plugin-node-resolve";
import {
  //babel,
  getBabelOutputPlugin,
} from "@rollup/plugin-babel";
import commonjs from "@rollup/plugin-commonjs";
import { terser } from "rollup-plugin-terser";

export default {
  input: "index.js",
  output: {
    file: "./dist/index.js",
    format: "cjs",
    exports: "named",
    sourcemap: process.env.NODE_ENV === "production" ? false : "inline",
    strict: false,
  },

  //https://dev.to/plebras/setting-up-a-javascript-build-process-using-rollup-n1e
  plugins: [
    auto(),
    resolve(),
    commonjs(),
    //babel()
    getBabelOutputPlugin({
      //"presets": ["@babel/env"],
      presets: [
        [
          "@babel/preset-env",
          {
            targets: {
              node: 10,
            },
            //modules: 'cjs'
          },
        ],
      ],
      plugins: ["add-module-exports"],
    }),
    terser({
      ecma: 2015,
      mangle: { toplevel: true },
      compress: {
        toplevel: true,
        drop_console: true,
        drop_debugger: true,
      },
      output: { quote_style: 1 },
    }),
  ],
};
